WordPress/src/main/java/org/wordpress/android/ui/DeepLinkingIntentReceiverViewModel.kt

```diff
@@ -29,117 +26,83 @@ class DeepLinkingIntentReceiverViewModel
                 } else {
-                    ShowSignInFlow
-                }
-                WP_LOGIN -> {
-                    buildNavigateAction(redirectUri)
+                    null
                 }
-                else -> null
             }
-        } else {
-            null
+            editorLinkHandler.isEditorUrl(uri) -> editorLinkHandler.buildOpenEditorNavigateAction(uri)
+            statsLinkHandler.isStatsUrl(uri) -> statsLinkHandler.buildOpenStatsNavigateAction(uri)
+            startLinkHandler.isStartUrl(uri) -> startLinkHandler.buildNavigateAction()
+            else -> null
+        }?.also {
+            // The new URL was build so we need to hit the original `mbar` tracking URL
+            if (trackingUrl) serverTrackingHandler.request(uri)
         }
-    }
-
-    private fun redirectToBrowser(uri: UriWrapper): NavigateAction {
-        return OpenInBrowser(uri.copy(REGULAR_TRACKING_PATH))
-    }
-
-    /**
-     * Opens post editor for provided uri. If uri contains a site and a postId
-     * (e.g. https://wordpress.com/post/example.com/1231/), opens the post for editing, if available.
-     * If the uri only contains a site (e.g. https://wordpress.com/post/example.com/ ), opens a new post
-     * editor for that site, if available.
-     * Else opens the new post editor for currently selected site.
-     */
-    fun handleOpenEditor(uri: UriWrapper) {
-        _navigateAction.value = Event(editorLinkHandler.buildOpenEditorNavigateAction(uri))
-    }
-
-
-    /**
-     * Opens stats screen for provided uri. If uri contains a stats period and site URL
-     * (e.g. https://wordpress.com/stats/day/example.com/)
-     * If the siteURL is not found for the current user, the link opens the stats screen for the current site
-     */
-    fun handleShowStats(uri: UriWrapper) {
-        _navigateAction.value = Event(statsLinkHandler.buildOpenStatsNavigateAction(uri))
+        return navigateAction ?: OpenInBrowser(
+                rootUri.copy(REGULAR_TRACKING_PATH)
+        )
     }
 
     private fun getRedirectUri(uri: UriWrapper): UriWrapper? {
         return deepLinkUriUtils.getUriFromQueryParameter(uri, REDIRECT_TO_PARAM)
     }
 
-    private fun shouldShow(uri: UriWrapper, path: String): Boolean {
-        return uri.host == HOST_WORDPRESS_COM && uri.pathSegments.firstOrNull() == path
-    }
-
     override fun onCleared() {
         serverTrackingHandler.clear()
         super.onCleared()
     }
 
     companion object {
-        private const val HOST_WORDPRESS_COM = "wordpress.com"
+        const val HOST_WORDPRESS_COM = "wordpress.com"
         private const val HOST_API_WORDPRESS_COM = "public-api.wordpress.com"
         private const val MOBILE_TRACKING_PATH = "mbar"
         private const val REGULAR_TRACKING_PATH = "bar"
         private const val REDIRECT_TO_PARAM = "redirect_to"
-        private const val POST_PATH = "post"
-        private const val STATS_PATH = "stats"
```

Comment:
I think we can also remove the `STATS_PATH` constant from `DeepLinkingIntentReceiverActivity`.

Reply:
fixed in 
8990aa1